src/pip/_internal/metadata/importlib/_dists.py

```diff
@@ -169,8 +170,10 @@ def _get_dist_name_from_location(self) -> Optional[str]:
 
     @property
```

Comment:
`@cached_property` would have been better but not usable until cpython 3.12 due to the undocumented locking behavior